### PR TITLE
REFACTOR: Improve compute power

### PR DESCRIPTION
**Pull request by hui-zhou-a**
## Description
**Please provide a brief description of the changes made in this pull request.**

## Issue linked
**Please mention the issue number or describe the problem this pull request addresses.**

## Checklist
- [x] I have tested my changes locally.
- [x] I have added necessary documentation or updated existing documentation.
- [x] I have followed the coding style guidelines of this project.
- [x] I have added appropriate tests (unit, integration, system).
- [x] I have reviewed my changes before submitting this pull request.
- [x] I have linked the issue or issues that are solved by the PR if any.
- [x] I have agreed with the Contributor License Agreement ([CLA](https://developer.ansys.com/form/cla-acceptance)).

### CODE REVIEW

**Comment by hui-zhou-a** [comment]
@Samuelopez-ansys @maxcapodi78 please help to review and merge this PR.

File: src/ansys/aedt/core/application/analysis_3d_layout.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def post(self):
 
         Returns
         -------
-        :class:`ansys.aedt.core.visualization.post.post_common_3d.PostProcessor3D`
+        :class:`ansys.aedt.core.visualization.post.post_processor`
```
Comment (maxcapodi78) [inline src/ansys/aedt/core/application/analysis_3d_layout.py:150]: why did you change the return class with a module?

File: src/ansys/aedt/core/visualization/post/post_3dlayout.py [hunk 2]
```diff
@@ -43,43 +46,113 @@ class PostProcessor3DLayout(PostProcessor3D):
...
+
+                loss = self._app.post.fields_calculator.evaluate(my_expression["name"], solution, intrinsics={})
+
+                power_loss_per_layer.append({"layer": layer_name, "net": net_name, "loss": float(loss)})
+
+        return power_loss_per_layer
 
     @pyaedt_function_handler()
-    def compute_power_by_layer(self, layers=None, nets=None, solution=None):
+    def compute_power_by_layer(self, solution=None):
```
Comment (maxcapodi78) [inline src/ansys/aedt/core/visualization/post/post_3dlayout.py:151]: I would have the possibility to filter by layer or net in any case (even if now is much faster). On big board could simplify the post-processing

File: src/ansys/aedt/core/visualization/post/post_3dlayout.py [hunk 3]
```diff
@@ -88,68 +161,23 @@ def compute_power_by_layer(self, layers=None, nets=None, solution=None):
...
+            loss = i["loss"]
+            if layer_name not in power_by_layers:
+                power_by_layers[layer_name] = loss
+            else:
+                power_by_layers[layer_name] += loss
         return power_by_layers
 
     @pyaedt_function_handler()
-    def compute_power_by_nets(self, nets=None, layers=None, solution=None):
+    def compute_power_by_net(self, solution=None):
```
Comment (maxcapodi78) [inline src/ansys/aedt/core/visualization/post/post_3dlayout.py:176]: Same as above. I would also leave the filter solution (with None as default) to be able to compute for the entire board or only for the selected layers/nets

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]
Few suggestions and comments

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Pull request merged**